content/essentials/components-and-props.mdx

```diff
@@ -0,0 +1,360 @@
+
+The lifecycle of a Solid component ends once it has been initially run, meaning it will not be re-run when the application's state changes.
+This is different from frameworks like React and Angular, where functions are defined and run with every application state change.
+
+With a Solid component's initial run, it sets up a reactive system. 
+This means that it monitors for specific state changes and makes adjustments to the relevant portions without the need to re-execute the entire component.
+By bypassing the full component lifecycle on every state change, Solid has a more predictable behavior compared to frameworks that re-run functions on every update, like React. 
+
+This approach emphasizes the importance of correctly establishing the initial reactive connections.
+Since the component logic isn't continuously visited, getting this setup right is crucial when working with Solid.
+To learn more about the concept of reactivity, refer to [intro to reacticity](/essentials/intro-to-reacticity)
+For a more comprehensive understanding of Solid's unique take on reactivity, explore this guide on [understanding fine-grained reactivity](/guide/understanding-fine-grained-reacticity).
+
+### Initialization & configuration
+
+During the first run of your component, Solid ensures the setup of reactive dependencies, event listeners, and other configurations. 
+This initial setup phase is crucial, not just for immediate functionality, but for the long-term reusability of your component. 
+This step defines how your component will interact with data and other elements.
+
+Solid components are designed with reusability in mind. 
+The initial configuration means that once a component is set up, it can be reused in different contexts without needing frequent reconfigurations. 
+This plays a pivotal role throughout the component's lifecycle.
+
+Having the proper connections and dependencies established from the outset optimizes performance and makes it easier to reuse components across various parts of an application or even across different projects. 
+
+```tsx
+function MyComponent() {
+  const [count, setCount] = createSignal(0);
+  
+  createEffect(() => console.log(count())) // 
+  return (
+    <div>
+      <p>Count: {count()}</p>
+      <button onClick={() => setCount(count() + 1)}>Increment</button>
+    </div>
+  );
+}
+```
+
+In this example, when `MyComponent` is used  (i.e., when it's rendered into the DOM for the first time), the function body is executed. 
+This includes the creation of the `count` signal and the `console.log(count())` statement.
+Once these lines are run, they will not be executed again for the life of this component instance, meaning the log statement will show only *once* in the console.
+
+The `return` statement provides JSX, which describes the structure and behavior of the component. 
+The actual rendering part (what's inside the `return`), however, is reactive. 
```

Comment:
Anything inside JSX, not anything after return. `let x = <>{props.x}</>` is reactive but `return props.x` is not.